aocd/runner.py

```diff
@@ -293,16 +329,16 @@ def run_one(
 
 
 def run_for(
-    plugs,
-    years,
-    days,
-    datasets,
-    example=False,
-    timeout=DEFAULT_TIMEOUT,
-    autosubmit=True,
-    reopen=False,
-    capture=False,
-):
+    plugs: Iterable[str],
```

Comment:
Plugs can't be an iterable, because it needs to support containment *and* iteration.

Using `plugs: Iterable[str]` here is wrong because you can't just pass in a generator yielding plugs to this function and expect it to work. Generators are valid iterables (they implement `__iter__`), but using a generator would lead to unexpected side effects. Further down, the line `eps = {ep.name: ep for ep in get_plugins() if ep.name in plugs}` uses a containment test on `plugs`, and if `plugs` is a generator then the `in` test would iterate over the values until it found a match. And that in turn leads to the `itertools.product()` function being passed a partially or fully iterated `plugs` iterator. Try passing in `(ep.name for ep in get_plugins())` for example; all the plugins would end up being skipped and no 

Use `typing.Collection[str]` instead:

```suggestion
    plugs: Collection[str],
```

Collections are iterable and support containment directly. Generators are not collections.